### DISCUSSION TITLE
알림 관련 API에 대해서

### DISCUSSION THREAD

**Discussion by hozzijeong** [woowacourse-teams/2023-pium#416] [개발 논의]
# 안녕하세요?

요약: 클린과 함께 push알림 구현할 백엔드 구함.

---
전문

다들 풍성한 한가위를 지내고 계신가요? 

다름이 아니라 푸시 메세지 관련 API가 필요해서 이렇게 discussion을 작성해 봅니다.

push 알림에 대한 대략적인 그림은 아래와 같습니다.

![image](https://github.com/woowacourse-teams/2023-pium/assets/50974359/974d0427-f153-49c5-9120-f054f42f8694)

firebase라는 push service를 통해서 push 알림을 구현하고자 합니다. 여기서 firebase에 구독을 신청할 때 기존에 firebase에 설정한 publicKey를 통해서만 구독 설정이 가능합니다. 하지만 이 publicKey를 클라이언트에서 갖고 있자니 약간 위험할 수도 있다는 생각에 서버에서 보내주는 방식으로 진행하려고 합니다. (근데, subModule에 넣어놔도 되지 않을까? 하는 생각이 있어서 이 부분은 선택사항이긴 합니다) 

이제, 해당 사용자가 리마인더 알림을 등록했다면 해당 사용자의 DB에 구독 요청/취소하는 속성(?)이 추가될 것 같습니다. 따라서 사용자가 구독을 취소하는 API와 구독을 요청하는 API가 필요할 것 같습니다.

여기에 더해서 저희가 알림을 보내길 원하는 기준(아마 리마인더가 되겠죠?)을 설정하고, 그에 맞춰서 알림을 push-service로 보내는? 작업도 같이 해야합니다!

### 요구 사항
-- 협업 사항 --
- [ ] public key를 반환해주는 API
- [ ] 구독을 취소하는 API
- [ ] 구독을 요청하는 API

-- 어떤 알림을 줄 지 설정 (아마 리마인더 알림일 될 듯) --
- [ ] 리마인더 알림을 날짜가 되면 보내주기

우선 제가 이해한 요구사항들은 위와 같습니다. 아마 함께 하게 되면서 유지되거나 변경될 수도 있습니다. (저도 잘 아는게 아니라서 흑흑)

--- 

### 참고 자료
우테코 4기 글입니당. https://wonsss.github.io/PWA/web-push-notification/#2-2-%EA%B5%AC%EB%8F%85-%EA%B3%BC%EC%A0%95

**Comment by Choi-JJunho** [comment]
저요저요 저 하고싶어요~

**Comment by hozzijeong** [comment]
### -----------마감-----------

**Comment by Choi-JJunho** [comment]
개발 완료로 close 하겠습니다.

### 백엔드
- https://github.com/woowacourse-teams/2023-pium/pull/430

### 프론트엔드
- https://github.com/woowacourse-teams/2023-pium/pull/469
- https://github.com/woowacourse-teams/2023-pium/pull/429
